kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -249,5 +414,88 @@ public interface TestActor {
+            }
+            else {
+                return new ResourceBuilder<>(latest, latest.getMetadata(), this);
+            }
+        }
+
+        default KafkaProtocolFilterResourceBuilder protocolFilter() {
+            return protocolFilter(randomName());
+        }
+
+        default KafkaProtocolFilterResourceBuilder protocolFilter(String name) {
+            KafkaProtocolFilter latest = get(KafkaProtocolFilter.class, name);
+            if (latest == null) {
+                return new KafkaProtocolFilterResourceBuilder(new KafkaProtocolFilterBuilder().withNewSpec().endSpec().build(),
+                        new ObjectMetaBuilder().withName(name).build(),
+                        this);
+            }
+            else {
+                return new KafkaProtocolFilterResourceBuilder(latest, latest.getMetadata(), this);
+            }
+        }
+
+        default VirtualKafkaClusterResourceBuilder virtualKafkaCluster() {
+            return virtualKafkaCluster(randomName());
+        }
+
+        default VirtualKafkaClusterResourceBuilder virtualKafkaCluster(String name) {
+            VirtualKafkaCluster latest = get(VirtualKafkaCluster.class, name);
+            if (latest == null) {
+                return new VirtualKafkaClusterResourceBuilder(new VirtualKafkaClusterBuilder().withNewSpec().endSpec().build(),
+                        new ObjectMetaBuilder().withName(name).build(),
+                        this);
+            }
+            else {
+                return new VirtualKafkaClusterResourceBuilder(latest, latest.getMetadata(), this);
+            }
+        }
+
+        default KafkaIngressResourceBuilder ingress() {
+            return ingress(randomName());
+        }
+
+        default KafkaIngressResourceBuilder ingress(String name) {
+            KafkaProxyIngress latest = get(KafkaProxyIngress.class, name);
+            if (latest == null) {
+                return new KafkaIngressResourceBuilder(new KafkaProxyIngressBuilder().withNewSpec().endSpec().build(), new ObjectMetaBuilder().withName(name).build(),
+                        this);
+            }
+            else {
+                return new KafkaIngressResourceBuilder(latest, latest.getMetadata(), this);
+            }
+        }
+
+        default KafkaServiceResourceBuilder kafkaService() {
+            return kafkaService(randomName());
+        }
+
+        default KafkaServiceResourceBuilder kafkaService(String name) {
+            KafkaService latest = get(KafkaService.class, name);
+            if (latest == null) {
+                return new KafkaServiceResourceBuilder(new KafkaServiceBuilder().withNewSpec().endSpec().build(), new ObjectMetaBuilder().withName(name).build(),
+                        this);
+            }
+            else {
+                return new KafkaServiceResourceBuilder(latest, latest.getMetadata(), this);
+            }
+        }
+
+    }
+
+    private static @NonNull String randomName() {
```

Comment:
The thing about random names is if the test fails it can be really hard (if you have logging) to impossible (if you don't) understanding what _actually_ went wrong. If the names are statically known then it's meaningful to be told that the `foo` didn't exist. If you're told the `curious ferret` didn't exist then you might not know how the `curious ferret` relates to any of the other things, and therefore have no idea what happened.

Reply:
hmm, I wonder if it would be an okay compromise to have some idea of 'groups' for the test, so I would create a kafkaProxy, VKC, service etc passing along a string like "foo" and that would be a prefix in the name. It seems like that's our pattern, loosely related groups of things.

Maybe the tests should be dealing in names though since it's a core aspect of what we're working with.